### DISCUSSION TITLE
Types like in Go

### DISCUSSION THREAD

**Discussion by GiorgosAlexakis** [carbon-language/carbon-lang#1496] [Language design]
Types in the form: float32,int8,int16 (similar to Go) would make the code more readable. 

Seeing var i: i64 = -1; looks very weird tbh. 

Why not:

var i int64 = -1;
or 
var i : int64 = -1;

Why does : have to be sticked on the left and not spaced out?
Or at least:

var i: int64 = -1;

I am not familiar with language design obviously ,but I am genuinely curious.

**Comment by GiorgosAlexakis** [comment]
Disliking it does not provide any useful feedback.

**Comment by robiot** [comment]
i64 is itself pretty easy to understand. A 64 bit integer. I don't think many developers would like to write it out like integer64.

Since the thing after the colon is what the type is, it makes sense to have it together with the variable than a space between. Many languages does it this way, Rust and Typescript just to name a few.

**Reply by GiorgosAlexakis** [reply]
Okay, that makes sense thank you.

**Reply by jonmeow** [reply]
> Types in the form: float32,int8,int16 (similar to Go) would make the code more readable.

For reference, the numeric types [match Rust](https://doc.rust-lang.org/reference/types/numeric.html). Part of the thought here is that developers frequently like short names, and Rust offers some evidence that the short name works for developers.

> Why does : have to be sticked on the left and not spaced out?

Spacing around the `:` is stylistic, not a requirement. The style is just leaning towards no space before, similar to how `:` is spaced in sentences.
